File: 2269 find-the-k-beauty-of-a-number/solution.cs
public class Solution {
    public int DivisorSubstrings(int num, int k) {
        string st = num.ToString();

        int count=0;

        for(int i=0;i<=st.Length-k;i++) {
            string sub = st.Substring(i,k);
            int v=int.Parse(sub);

            if(v!=0 && num%v==0) count++;
        }
        return count;
    }
}

File: 2269 find-the-k-beauty-of-a-number/solution.rs
impl Solution {
    pub fn divisor_substrings(num: i32, k: i32) -> i32 {
        let str = num.to_string();
        let mut c=0;
        let k=k as usize;

        for i in 0..str.len()-k+1 {
            let sub = &str[i..i+k];
            let v:i32=sub.parse().unwrap();
            if v!=0 && num%v==0 {
                c+=1;
            }
        }
        c
    }
}
